src/buffer/out/textBuffer.cpp

```diff
@@ -1033,238 +1036,172 @@ const TextBuffer::TextAndColor TextBuffer::GetTextForClipboard(const bool lineSe
-                    szSpanStart.resize(cbSpanStart); // chop null from sprintf
-                    szClipboard.append(szSpanStart);
-                }
+                    if (hasWroteAnyText)
+                    {
+                        htmlBuilder << "</SPAN>";
+                    }
 
-                // accumulate 1 character
-                cchCharsToPrint++;
-            }
+                    htmlBuilder << "<SPAN STYLE=\"";
+                    htmlBuilder << "color:";
+                    htmlBuilder << Utils::ColorToHexString(fgColor.value());
+                    htmlBuilder << ";";
+                    htmlBuilder << "background-color:";
+                    htmlBuilder << Utils::ColorToHexString(bkColor.value());
+                    htmlBuilder << ";";
+                    htmlBuilder << "\">";
+                }
 
-            PCWCHAR pwchAccumulateStart = rows.text.at(iRow).data() + cbStartOffset;
+                hasWroteAnyText = true;
 
-            // write accumulated characters to stream
-            std::string CharsConverted;
-            int cbCharsConverted = WideCharToMultiByte(CP_UTF8, 0, pwchAccumulateStart, static_cast<int>(cchCharsToPrint), nullptr, 0, nullptr, nullptr);
-            CharsConverted.resize(cbCharsConverted);
-            WideCharToMultiByte(CP_UTF8, 0, pwchAccumulateStart, static_cast<int>(cchCharsToPrint), CharsConverted.data(), cbCharsConverted, nullptr, nullptr);
-            szClipboard.append(CharsConverted);
+                if (isLastCharInRow)
+                {
+                    writeAccumulatedChars(true);
+                    break;
+                }
+            }
         }
 
-        if (bColorFound)
+        if (hasWroteAnyText)
         {
-            // copy end span
-            szClipboard.append(szSpanEnd);
+            // last opened span wasn't closed in loop above, so close it now
+            htmlBuilder << "</SPAN>";
         }
 
-        // after we have copied all text we must wrap up
-        // with a standard set of HTML boilerplate required
-        // by CF_HTML
-
-        // copy end font size span
-        szClipboard.append(szSpanEnd);
-
-        // copy end font face span
-        szClipboard.append(szSpanEnd);
+        htmlBuilder << "</DIV>";
 
-        // copy end background color span
-        szClipboard.append(szDivEnd);
+        htmlBuilder << "<!--EndFragment -->";
 
-        // copy HTML end fragment
-        szClipboard.append(szHtmlFragEnd);
+        constexpr std::string_view HtmlFooter = "</BODY></HTML>";
+        htmlBuilder << HtmlFooter;
 
-        // copy HTML footer
-        szClipboard.append(szHtmlFooter);
+        // once filled with values, there will be exactly 157 bytes in the clipboard header
+        constexpr size_t ClipboardHeaderSize = 157;

```

Comment:
Why 157? It'd probably be best to calculate this value in case we ever update the header.

Reply:
This is chicken-and-egg problem. We have to write header's size in header, so it must be known before we write the header. Unless we wrote the actual html at offset 1000 and hoped that header won't ever be longer than 1000.

Reply:
Good enough for me. #Resolved